### PR TITLE
Clarify the meaning of boolean return values

### PR DESCRIPTION
**Pull request by messmerd**
Many functions that return `bool` do not specify what that value means, so I marked those functions with "Returns true on success" or similar wording.

Along the way I also fixed a few typos and style issues.

### CODE REVIEW

File: include/clap/ext/timer-support.h [hunk 1]
```diff
@@ -16,10 +16,11 @@ typedef struct clap_plugin_timer_support {
 typedef struct clap_host_timer_support {
    // Registers a periodic timer.
    // The host may adjust the period if it is under a certain threshold.
-   // 30 Hz should be allowed.
+   // 30 Hz should be allowed. Returns true on success.
```
Comment (abique) [inline include/clap/ext/timer-support.h:19]: Maybe line break here?

File: include/clap/ext/note-ports.h [hunk 2]
```diff
@@ -40,11 +40,12 @@ typedef struct clap_note_port_info {
 
 // The note ports scan has to be done while the plugin is deactivated.
 typedef struct clap_plugin_note_ports {
-   // number of ports, for either input or output
+   // Number of ports, for either input or output
```
Comment (abique) [inline include/clap/ext/note-ports.h:43]: If you start with capital, then end with `.` maybe?

File: include/clap/ext/draft/triggers.h [hunk 3]
```diff
@@ -88,7 +88,6 @@ typedef struct clap_plugin_triggers {
    uint32_t(CLAP_ABI *count)(const clap_plugin_t *plugin);
 
    // Copies the trigger's info to trigger_info and returns true on success.
-   // Returns true on success.
```
Comment (abique) [inline include/clap/ext/draft/triggers.h:91]: Why did you remove that one?

**Comment by abique** [comment]
Thank very much you for this change, please address the comments and it should be good :+1:

File: include/clap/ext/draft/triggers.h (hunk 3 above)
Comment (messmerd) [reply include/clap/ext/draft/triggers.h:91]: The line above it already mentioned "returns true on success"

Comment (abique) [reply include/clap/ext/draft/triggers.h:91]: Ah correct!
